### PR TITLE
Feature/cp 3416/global state redux

### PR DESCRIPTION
**Pull request by nf2000**
Redux introduced 

https://chillibean.atlassian.net/browse/CP-3416

### CODE REVIEW

File: nims-notes-things/src/components/Form.js [hunk 1]
```diff
@@ -17,6 +21,13 @@ const Form = (props) => {
     note: false,
   });
 
+  const dispatch = useDispatch();
+
+  const notes = useSelector((state) => {
+    console.log(state)
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove

File: nims-notes-things/src/components/Form.js [hunk 2]
```diff
@@ -17,6 +21,13 @@ const Form = (props) => {
...
+    return state.noteReducer.value;
+  })
+  console.log(notes);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove

File: nims-notes-things/src/components/Form.js [hunk 3]
```diff
@@ -3,6 +3,10 @@ import styled from "styled-components";
 import TextArea from "./TextArea";
 import Input from "./Input";
 import Button from "./Button";
+// import { v4 as uuidv4 } from "uuid";
+import { useSelector } from "react-redux";
+import { useDispatch } from "react-redux";
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: you can put useSelector and useDispatch in the same line
`import { useSelector, useDispatch } from "react-redux";`

File: nims-notes-things/src/components/Form.js [hunk 4]
```diff
@@ -25,6 +36,40 @@ const Form = (props) => {
...
+    Object.keys(formData).forEach((data) => {
+      if (formData[data] === "") {
+        setFormEmpty((previousState) => {
+          return {
+            ...previousState,
+            [data]: true,
+          };
+        });
+        return (isValidated = false);
+      } else {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js:50]: No need for else statement here, see Alex's

File: nims-notes-things/src/components/Form.js [hunk 5]
```diff
@@ -25,6 +36,40 @@ const Form = (props) => {
...
+    let isValidated = true;
+    Object.keys(formData).forEach((data) => {
+      if (formData[data] === "") {
+        setFormEmpty((previousState) => {
+          return {
+            ...previousState,
+            [data]: true,
+          };
+        });
+        return (isValidated = false);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js:49]: Also no need to return anything... just need to put `isValidated = false`

File: nims-notes-things/src/components/Form.js [hunk 6]
```diff
@@ -25,6 +36,40 @@ const Form = (props) => {
...
+    return isValidated;
+  };
+
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    const isValidated = validateSubmit();
+    if (isValidated) {
+      const formdatacopy = {...formData}
+      const notesCopy = [...notes];
+      //console.log(formdatacopy);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove comments and consoles

File: nims-notes-things/src/components/Form.js [hunk 7]
```diff
@@ -25,6 +36,40 @@ const Form = (props) => {
...
+      //console.log(notesCopy);
+      console.log(notesCopy);
+
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove whitespace

File: nims-notes-things/src/components/Form.js [hunk 8]
```diff
@@ -25,6 +36,40 @@ const Form = (props) => {
...
+      }
+    });
+    return isValidated;
+  };
+
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    const isValidated = validateSubmit();
+    if (isValidated) {
+      const formdatacopy = {...formData}
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: this is where you can add a unique id with `uuid`.
```
const uniqueId = uuidv4();
const formDataCopy = {
   ...formData,
   id: uniqueId
}
```

Comment (damonspencerroberts) [reply nims-notes-things/src/components/Form.js]: also make sure to always use camel case `formdatacopy` should be `formDataCopy`

Comment (damonspencerroberts) [reply nims-notes-things/src/components/Form.js]: also make sure to format document...

File: nims-notes-things/src/components/notes.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    return state.noteReducer.value;
+  });
+
+  return (
+    <div>
+      {notes.map((n) => {
+        return (
+          <div>
+            <Note>
+              <pa>{n.name}</pa>
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: `<pa />`?

File: nims-notes-things/src/redux/store.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { createStore, applyMiddleware, compose } from "redux";
+import thunk from "redux-thunk";
+import rootReducer from "./reducers/rootReducer";
+
+const initialState = {};
+const middleware = [thunk];
+
+// if you want to look at the store ensure you have the redux state extension installed on chrome to see the redux state
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/store.js]: you can remove this comment

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Couple things here. Try to remember to remove comments, console.logs and also format documents before review

File: nims-notes-things/src/components/notes.js (hunk 9 above)
Comment (nf2000) [reply nims-notes-things/src/components/notes.js]: that was just me playing around with style component - i forgot to remove it

**Review by damonspencerroberts — APPROVED** [review]
Approved!

**Pull request merged**